src/NServiceBus.Callbacks/SynchronousOptions.cs

```diff
@@ -0,0 +1,53 @@
+namespace NServiceBus
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    ///     Allows the users to control how the send is performed
+    /// </summary>
+    public class SynchronousOptions
```

Comment:
Just noting: With this we do not longer support callbacks for deferred messages.

Ie: bus.Defer(new MyMessage()).Register() //not supported anymore